assets/js/base/context/cart-checkout/event-emit/emitters.js

```diff
@@ -24,17 +24,21 @@ const getObserversByPriority = ( observers, eventType ) => {
  */
 export const emitEvent = async ( observers, eventType, data ) => {
 	const observersByType = getObserversByPriority( observers, eventType );
+	let emitterResponse = {};
 	for ( const observer of observersByType ) {
 		try {
-			await Promise.resolve( observer.callback( data ) );
+			const response = await Promise.resolve( observer.callback( data ) );
+			if ( typeof response === 'object' ) {
+				emitterResponse = { ...emitterResponse, ...response };
```

Comment:
This change is in order to collect errors produced by all observers without aborting on the first one that fails. This allows us to show validation errors + payment method errors at once. I looked at how `emitEvent` is used around the codebase and I think this change should be safe, but would like some feedback if somebody thinks it will break something now or in the future.

Reply:
Looks okay to me.